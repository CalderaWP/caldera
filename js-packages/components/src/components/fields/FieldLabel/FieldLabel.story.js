import React from 'react';
import { storiesOf } from '@storybook/react';
import { FieldLabel } from './FieldLabel';

storiesOf('FieldLabel', module).add('Is a label', () => (
	<FieldLabel
		fieldType={'text'}
		fieldId={'text-field-test'}
	>Label For Text Field</FieldLabel>
));
