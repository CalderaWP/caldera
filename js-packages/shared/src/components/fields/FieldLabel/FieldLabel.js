import {
	labelClassNames
} from '../util';

import React from 'react'

export const FieldLabel = ( { fieldType, children, fieldId } ) => {
	return (
		<label
			htmlFor={fieldId}
			className={labelClassNames(fieldType)}
		>
			{children}
		</label>
	);
};

