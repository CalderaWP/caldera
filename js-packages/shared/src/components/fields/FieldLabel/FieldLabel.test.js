import * as React from 'react';
import renderer from 'react-test-renderer';

import { FieldLabel } from './FieldLabel';

test('Field label', () => {
	const component = renderer.create(
		<FieldLabel
			fieldType={'text'}
			fieldId={'hello'}
			>
			Hi Roy
		</FieldLabel>
			);
	expect(component.toJSON()).toMatchSnapshot();
});
