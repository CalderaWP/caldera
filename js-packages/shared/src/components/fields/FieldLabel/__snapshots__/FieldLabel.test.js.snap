// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field label 1`] = `
<label
  className="caldera-field-label caldera-field-label-text"
  htmlFor="hello"
>
  Hi Roy
</label>
`;
